### PR TITLE
bank: verify precompiles for transactions

### PR DESCRIPTION
**Pull request by ptaffet-jump**
Fixes [#1438](https://github.com/firedancer-io/firedancer/issues/1438)
Though pack's cost estimation should be updated to work with them

### CODE REVIEW

File: src/disco/metrics/metrics.xml [hunk 1]
```diff
@@ -213,6 +213,8 @@ metric introduced.
     <counter name="TransactionSanitizeFailure" summary="Number of transactions that failed to sanitize." />
     <counter name="TransactionNotExecutedFailure" summary="Number of transactions that did not execute. This is different than transactions which fail to execute, which make it onto the chain." />
 
+    <counter name="PrecompileVerifyFailed" summary="Number of transactions that failed precompile verification and thus will not execute." />
```
Comment (mmcgee-jump) [inline src/disco/metrics/metrics.xml:216]: nit: maybe keep naming consistent with the above ones, `PrecompileVerifyFailure`

File: src/app/fdctl/run/tiles/fd_bank.c [hunk 2]
```diff
@@ -182,6 +183,12 @@ after_frag( void *             _ctx,
     ctx->metrics.txn_load_address_lookup_tables[ result ]++;
     if( FD_UNLIKELY( result!=FD_BANK_ABI_TXN_INIT_SUCCESS ) ) continue;
 
+    int precompile_result = fd_ext_bank_verify_precompiles( ctx->_bank, abi_txn );
+    if( FD_UNLIKELY( precompile_result ) ) {
+      FD_MCNT_INC( BANK_TILE, PRECOMPILE_VERIFY_FAILED, 1 );
```
Comment (mmcgee-jump) [inline src/app/fdctl/run/tiles/fd_bank.c:188]: nit: could be a lot of cache traffic if someone spams txns that fail precompiles, they also don't pay right

**Review by mmcgee-jump — DISMISSED** [review]

File: src/app/fdctl/run/tiles/fd_bank.c (hunk 2 above)
Comment (ptaffet-jump) [reply src/app/fdctl/run/tiles/fd_bank.c:188]: I think each precompile takes tens of microseconds of computation, so the cache traffic will be in the noise

File: src/disco/metrics/metrics.xml (hunk 1 above)
Comment (ptaffet-jump) [reply src/disco/metrics/metrics.xml:216]: Good call. I renamed it.

**Review by mmcgee-jump — APPROVED** [review]

**Pull request merged**